### PR TITLE
Support uuids

### PR DESCRIPTION
**Pull request by nanu-c**
This needs the corresponding branch to be merged https://github.com/signal-golang/textsecure/tree/groupsv2

### CODE REVIEW

File: app/store/chatList.go [hunk 1]
```diff
@@ -230,61 +244,112 @@ func UpdateTimestamps() {
...
+func (s *Sessions) Get(id int64) (*Session, error) {
+	for _, ses := range s.Sess {
+		if ses.ID == id {
+			return ses, nil
+		}
+	}
+	return nil, fmt.Errorf("Session with id %d not found", id)
+}
+
+// GetByE164 returns the session by the telephone number amd creates it if it doesn't exists
```
Comment (Flaburgan) [inline app/store/chatList.go:263]: I don't see anything which create the session here, is that comment up-to-date?

File: app/store/db.go [hunk 2]
```diff
@@ -90,13 +92,20 @@ func (ds *DataStore) SetupDb(password string) bool {
 	UpdateMessagesTable_v_0_7_8()
 	UpdateSessionTable_v_0_7_8()
 	UpdateSessionTable_v_0_9_0()
+	UpdateSessionTable_v_0_9_3()
```
Comment (Flaburgan) [inline app/store/db.go:95]: it's 0.9.4 now ;)

File: app/store/migrations.go [hunk 3]
```diff
@@ -107,3 +107,46 @@ func UpdateSessionTable_v_0_9_0() error {
 
 	return err
 }
+
+// add support uuids
+func UpdateSessionTable_v_0_9_3() error {
```
Comment (Flaburgan) [inline app/store/migrations.go:112]: That's 0.9.4 now ;)

File: app/store/migrations.go [hunk 4]
```diff
@@ -107,3 +107,46 @@ func UpdateSessionTable_v_0_9_0() error {
...
+		return err
+	}
+	res, err := statement.Query()
+	if err != nil {
+		return err
+	}
+
+	col, err := res.Columns()
+	if len(col) == 10 {
+		log.Infof("[axolotl] Update sessions schema v_0_9_3")
```
Comment (Flaburgan) [inline app/store/migrations.go:125]: Ditto

File: app/store/migrations.go [hunk 5]
```diff
@@ -107,3 +107,46 @@ func UpdateSessionTable_v_0_9_0() error {
...
+	}
+	resM, err := statementM.Query()
+	if err != nil {
+		return err
+	}
+
+	colM, err := resM.Columns()
+
+	if len(colM) == 17 {
+		log.Infof("[axolotl] Update messages schema v_0_9_3")
```
Comment (Flaburgan) [inline app/store/migrations.go:144]: Ditto

File: app/ui/gui.go [hunk 6]
```diff
@@ -106,10 +84,7 @@ func runUIUbuntuTouch(e string) {
 		cmd = exec.Command("qmlscene", "--scaling", "guis/qml/ut/MainUt.qml")
 	} else if config.Gui == "qt" {
 		guilocation := os.Getenv("AXOLOTL_GUI_DIR")
-		cmd = exec.Command("qmlscene", "--scaling", guilocation + "guis/qml/qml/MainQML.qml")
-	} else if config.Gui == "me" {
```
Comment (Flaburgan) [inline app/ui/gui.go:110]: This change looks unrelated to the UUID integration, what is this about?

File: app/webserver/webserver.go [hunk 7]
```diff
@@ -134,28 +131,37 @@ func wsReader(conn *websocket.Conn) {
...
-			sendCurrentChat(s)
+
+			log.Println("[axolotl] Open chat with id: ", openChatMessage.Id)
+			s, err := store.SessionsModel.Get(openChatMessage.Id)
+			if err != nil {
+				log.Errorln("[axolotl] Open chat with id: ", openChatMessage.Id, "failed", err)
+			} else {
+				activeChat = openChatMessage.Id
+				if !s.IsGroup {
+					// p, _ := textsecure.GetProfile(s.Tel)
```
Comment (Flaburgan) [inline app/webserver/webserver.go:150]: Work to be done there? If so, write a detailed TODO please ;)

**Review by Flaburgan — CHANGES_REQUESTED** [review]
I read everything but I was just looking for typo / obvious error, I didn't try to understand it.
Are you already using that code for real? How can I test it?

File: app/store/chatList.go (hunk 1 above)
Comment (nanu-c) [reply app/store/chatList.go:263]: This is not for creating a session but to retrieve the an existing session.

File: app/store/chatList.go [hunk 8]
```diff
@@ -46,57 +50,64 @@ var SessionsModel = &Sessions{
...
-func (s *Sessions) GetMessageList(id string) (error, *MessageList) {
-	index := SessionsModel.GetIndex(id)
-	if index == -1 {
-		index = int(SessionsModel.Get(id).ID)
-	}
-	if index != -1 {
+
+// GetMessageList returns the message list for the session id
+func (s *Sessions) GetMessageList(ID int64) (error, *MessageList) {
+	if ID != -1 {
```
[the remark this answers is not in the document]
Comment (nanu-c) [reply app/store/chatList.go]: Done

File: app/store/chatList.go (hunk 1 above)
Comment (nanu-c) [reply app/store/chatList.go]: You were right, it's solved now.

File: app/webserver/webserver.go (hunk 7 above)
Comment (nanu-c) [reply app/webserver/webserver.go:150]: Done

File: app/handler/handler.go [hunk 9]
```diff
@@ -117,7 +117,28 @@ func buildAndSaveMessage(msg *textsecure.Message, syncMessage bool) {
 		msgFlags = helpers.MsgFlagReaction
 		text = msg.Reaction().GetEmoji()
 	}
-	session := store.SessionsModel.Get(s)
+	session, err := store.SessionsModel.GetByUUID(msgSource)
+	if err != nil && gr == nil {
+		// Session could not be found, lets try to find it by E164 aka phone number
+		log.Println("[axolotl] MessageHandler Error ", err)
+		session = store.SessionsModel.GetByE164(msg.Source())
+		if session != nil && gr == nil {
```
[the remark this answers is not in the document]
Comment (nanu-c) [reply app/handler/handler.go]: Nope, you are right.
